ring/batch.go

```diff
@@ -191,9 +190,11 @@ func (b *batchTracker) record(itemTrackers []*itemTracker, err error, isClientEr
 
 			// If we successfully called this particular instance, but we don't have any remaining instances to try,
 			// and we failed to call minSuccess instances, then we need to return the last error.
-			if it.remaining.Dec() == 0 {
-				if b.rpcsFailed.Inc() == 1 {
-					b.err <- it.err.Load()
+			if it.succeeded.Load() < int32(it.minSuccess) {
```

Comment:
isn't there a race between calling `it.suceeded.Inc()` and `it.succeeded.Load()`? It's possible that between the time one invocation of `succeeded.Inc()` gets to `succeeded.Load()` another one has already invoked `Inc()`. This was we lose one invocation of `remaining.Dec()` of the first one

you can save the return value of succeeded.Inc in a variable and use that instead of calling Load again

Reply:
That's correct. I will fix this error. Thank you @dimitarvdimitrov